### PR TITLE
Add cmake

### PR DESCRIPTION
**Pull request by rueyaa332266**
Create new PR after enable circleci for fork.
Build with cmake.
Fix README and circleci config.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,55 +16,61 @@ commands:
...
       - image: rhikimochi/opencv-docker:v0.14
     steps:
       - checkout
-      - prepare_for_build
+      - prepare_for_cmake
       - run:
-          name: Make object file
-          command: g++ -std=c++11 ./src/imageDiffCalc.cpp `pkg-config --libs --cflags opencv` -Iinclude -c
+          name: build gazosan
+          command: cd build && cmake .. -DGTEST=OFF && make && cd ..
```
Comment (hikimochi) [inline .circleci/config.yml]: -DGTEST=OFF is necessary?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  set (gtest ${CMAKE_SOURCE_DIR}/tests/googletest/libgtest.a)
+  set (gtest_main ${CMAKE_SOURCE_DIR}/tests/googletest/libgtest_main.a)
+  include_directories( src/ )
+  include_directories( include/ )
+  include_directories( tests/googletest/include/ )
+  file(DOWNLOAD https://raw.githubusercontent.com/jarro2783/cxxopts/master/include/cxxopts.hpp ${CMAKE_SOURCE_DIR}/include/cxxopts.hpp)
+
+  if( ${TEST_LEVEL} STREQUAL "unit" )
+    add_executable(gazosan_unit_test tests/unit_tests/imageDiffCalc_test.cpp)
+    target_link_libraries( gazosan_unit_test ${OpenCV_LIBS} ${gtest} ${gtest_main} -lpthread )
```
Comment (hikimochi) [inline CMakeLists.txt]: I want to commonalize.

```suggestion
    target_link_libraries( gazosan_unit_test ${LIBRARIES} )
```

```
set (LIBRALIES 
       ${OpenCV_LIBS}
       ${gtest} 
       ${gtest_main}
       -lpthread)
```

```

File: .circleci/build_googletest.sh [hunk 3]
```diff
@@ -1,11 +1,11 @@
 #!/bin/bash
```
Comment (hikimochi) [inline .circleci/build_googletest.sh:1]: I think the place of this shell script -> tests/ is better than .circleci/

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 2.8)
+project( Gazo-san )
+set (CMAKE_CXX_STANDARD 11)
+set(GTEST OFF CACHE BOOL "Test flag")
+find_package( OpenCV REQUIRED )
+include_directories( ${OpenCV_INCLUDE_DIRS} )
+
+if(GTEST)
```
Comment (hikimochi) [inline CMakeLists.txt:8]: Before build GTEST by cmake, we need running build_googletest.sh.
I think it is good to execute build_googletest.sh in this cmake if not exists google tests source.

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    add_executable( gazosan_system_test tests/system_tests/imageDiffCalc_test.cpp )
+    target_link_libraries( gazosan_system_test ${OpenCV_LIBS} ${gtest} ${gtest_main} -lpthread )
+  endif()
+else()
+  set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin )
+
+  if(EXISTS "${CMAKE_SOURCE_DIR}/libimageDiffCalc.a")
+    # Use static link library file
+    # Works only on ubuntu
+    add_executable( gazosan src/main.cpp )
```
Comment (hikimochi) [inline CMakeLists.txt]: please be variable -> gazosan because fix four times when we modify binary file name.

**Review by hikimochi — CHANGES_REQUESTED** [review]
Commented.

File: .circleci/config.yml (hunk 1 above)
Comment (rueyaa332266) [reply .circleci/config.yml]: It don't need to wright in here.
I will make it more shorter.

File: CMakeLists.txt (hunk 2 above)
Comment (rueyaa332266) [reply CMakeLists.txt]: cool

File: .circleci/build_googletest.sh (hunk 3 above)
Comment (rueyaa332266) [reply .circleci/build_googletest.sh:1]: I'll move the script

File: CMakeLists.txt (hunk 4 above)
Comment (rueyaa332266) [reply CMakeLists.txt:8]: Add execute shell script in cmake.

File: CMakeLists.txt (hunk 5 above)
Comment (rueyaa332266) [reply CMakeLists.txt]: I change the name as variable.

**Review by hikimochi — APPROVED** [review]
LGTM

**Pull request merged**